### PR TITLE
[sobjectizer] update to v.5.7.2.2

### PR DESCRIPTION
**Pull request by eao197**
SObjectizer updated to the latest version.

### CODE REVIEW

File: ports/sobjectizer/portfile.cmake [hunk 1]
```diff
@@ -1,8 +1,8 @@
 vcpkg_from_github(
     OUT_SOURCE_PATH SOURCE_PATH
     REPO stiffstream/sobjectizer
-    REF 04ec6fcf63eb977e9f45348b7c05b0d98c8f0b60 # v.5.7.2.1
-    SHA512 b2550d7bf7f3e405fccff02ae12fd2a235016c5c8bdcfc0bb1baf3a91f90369d68b2c57dbecbc990a8627338b7dcd6b3d32b681aaf0335dde0b2672003fe38ec
+    REF 73b0d1bf5b4bdf543c24bc4969d2c408e0cea812 # v.5.7.2.2
+    SHA512 f911900f67832be877ac1d906deb4fb03925601d91c7d9bb0dd1c789d71a5bddb8ec5e0fe361ad7cfa8787a70d09aee26b17e8dba55c0b5dc4fba2b31651442b
 )
 
 if(VCPKG_LIBRARY_LINKAGE STREQUAL static)
```
Comment (NancyLi1013) [inline ports/sobjectizer/portfile.cmake:8]: Could you please update this part as follows?

```
string(COMPARE EQUAL "${VCPKG_LIBRARY_LINKAGE}" "static" SOBJECTIZER_BUILD_STATIC )
string(COMPARE EQUAL "${VCPKG_LIBRARY_LINKAGE}" "dynamic" SOBJECTIZER_BUILD_SHARED)
```

Please see https://github.com/microsoft/vcpkg/blob/master/docs/maintainers/maintainer-guide.md#choose-either-static-or-shared-binaries

**Review by NancyLi1013 — CHANGES_REQUESTED** [review]

**Comment by NancyLi1013** [comment]
Hi @eao197 

Thanks for your PR.

The failures on x86-windows are like this:
```
Found the following errors:
Error: While reading versions for port sobjectizer from file: C:\a\1\s\versions\s-\sobjectizer.json
       Version `5.7.2.2` was not found in versions file.
       Run:

           vcpkg x-add-version sobjectizer

       to add the new port version.
```
Please run `vcpkg x-add-version sobjectizer` and then commit the changes.

Note: Please update and submit other changes, then run the above command.

File: ports/sobjectizer/portfile.cmake (hunk 1 above)
Comment (eao197) [reply ports/sobjectizer/portfile.cmake]: Thanks, I've updated this part.

**Comment by eao197** [comment]
Hi, @NancyLi1013 

I've updated the PR.

**Review by NancyLi1013 — APPROVED** [review]

**Comment by NancyLi1013** [comment]
The regressions caused by `gsoap ` will be fixed by PR #16270.

**Comment by ras0219-msft** [comment]
LGTM, thanks for the PR!

**Pull request merged**